### ISSUE TITLE
Metadata still being included in message when using Winston 3 

### ISSUE THREAD

**Issue by carsoni** [winstonjs/winston-mongodb#128]
I think the issue raised in #122 is still an active problem despite purportedly being corrected in winston-transport 3.2.0.

I am trying the following using winston 3.0.0 and winston-mongodb 4.0.0-rc1 with typescript

```
// tslint:disable-next-line:no-var-requires
const { config, createLogger, format, transports } = require('winston');

const myFormat = format.printf((info: any) => {
	return `${info.level} ${info.timestamp} [${info.label}] : ${info.message}`;
});

let logger = createLogger({
	levels: config.syslog.levels,
	level: 'debug',
	transports: [
		new transports.Console({
			format: format.combine(
				format.colorize(),
				format.timestamp({ format: 'YYYYMMDD HH:mm:ss.SSSS' }),
				format.label({ label: ' AUTHOR ' }),
				myFormat
			)
		})
	]
});

// tslint:disable:no-var-requires
// tslint:disable-next-line:no-unused-expression
require('winston-mongodb').MongoDB;

logger.add(
	new transports.MongoDB({
		db: 'mongodb://localhost/paradigm',
		collection: 'testLogging',
		level: 'debug',
		storeHost: true,
		format: format.combine(
			format.metadata({ fillExcept: ['level', 'message', 'timestamp', 'label'] }),
			format.metadata({ fillWith: ['engineRun', 'signalName', 'signalOutcome'], key: 'signalInfo' }),
			format.metadata({ fillWith: ['value', 'date', 'accountID'], key: 'accountInfo' }),
			format.json()
		),
		label: 'AUTHOR'
	})
);

logger.log('emerg', 'This message should be outside of the metadata objects', {
	engineRun: '6affc14f-86e3-4b22-b582-85e0b4d3e965',
	signalName: 'Erebor II',
	signalOutcome: true,
	value: 102765,
	date: '2018104 14:05:35',
	accountID: '42675'
});

```
In my case the metadata entry in the mongodb document always remains null and the message field contains

```
{"level":"emerg","message":"This message should be outside of the metadata objects","metadata":{"engineRun":"6affc14f-86e3-4b22-b582-85e0b4d3e965","signalName":"Erebor II","signalOutcome":true,"value":102765,"date":"2018104 14:05:35","accountID":"42675"},"signalInfo":{},"accountInfo":{}}

```

As you can see the message now also contains two empty objects "accountInfo" and "signalInfo" but nothing has been formatted or separated into objects.

If I revert to winston-mongodb@3.0.2 I get an upgrade warning "mongodb is a legacy winston transport. Consider upgrading:" and the separation is partially successful.

The message, timestamp, level are now all separated and while the metadata object as provided in the log call has been entered into the metadata field it has not been separated into the two required info objects and also includes the message and level data points as well - it looks like this

```
    {
	engineRun: '6affc14f-86e3-4b22-b582-85e0b4d3e965',
	signalName: 'Erebor II',
	signalOutcome: true,
	value: 102765,
	date: '2018104 14:05:35',
	accountID: '42675',
        level:  'emerg',
        message: 'This message should be outside of the metadata objects'
    }

```

Does anyone have any thoughts on this issue?
Many thanks in advance
Ian Carson

**Comment by carsoni** [comment]
@yurijmikhalevich Thanks Yurij. I've been trying different things to solve this all day and I hope I am not doing something stupid to stop it from working

**Comment by yurijmikhalevich** [comment]
@carsoni fixed in version 4.0.2.
